File: data/frama_c_problems/triangle_sides.c
#include <stdio.h>
#include <limits.h>
/*@
    requires a>0 && b>0 && c>0;
    requires a+b+c <= INT_MAX;
    behavior valid:
        assumes (a+b>c) && (a+c>b) && (b+c>a);
        ensures \result == 1;
    behavior invalid:
        assumes !((a+b>c) && (a+c>b) && (b+c>a));
        ensures \result == 0;
    disjoint behaviors;
    complete behaviors;
*/

int validts(int a, int b, int c) {
    int valid = 0;
    if ((a+b>c) && (a+c>b) && (b+c>a)) {
        valid = 1;
    } else {
        valid = 0;
    }
    return valid;
}

void test() {
    int valid = validts(2,3,4);
    //@ assert valid == 1;
}

File: transpiled_rust/frama_c_problems/triangle_sides.rs
fn validts(a: i32, b: i32, c: i32) -> i32 {
    let mut valid: i32 = 0;
    if (a + b > c) && (a + c > b) && (b + c > a) {
        valid = 1;
    } else {
        valid = 0;
    }
    valid
}

fn test() {
    let valid: i32 = validts(2, 3, 4);
    // verus_assert(1);
}